core/src/env/srml/types.rs

```diff
@@ -34,7 +34,13 @@ impl EnvTypes for DefaultSrmlTypes {
 
 /// The default SRML address type.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Encode, Decode)]
-pub struct Address(pub [u8; 32]);
+pub struct Address([u8; 32]);
+
+impl Address {
+    pub fn new(address: [u8; 32]) -> Address {
```

Comment:
For implementations like these it is recommended to just implement `From<[u8;32]>`.

Reply:
Same for `Hash`